pkg/streamingpromql/query.go

```diff
@@ -320,6 +330,8 @@ func (q *Query) Exec(ctx context.Context) *promql.Result {
 		return &promql.Result{Err: compat.NewNotSupportedError(fmt.Sprintf("unsupported result type %s", parser.DocumentedType(q.statement.Expr.Type())))}
 	}
 
+	q.result.Warnings = *q.annotations
```

Comment:
I thought Info was added upstream. Why not add support for it now?

Reply:
tl;dr: this is all we need to do here to cover both info and warning annotations

The API here is a bit confusing, but we're stuck with what Prometheus has done: `promql.Result` has a single field called `Warnings` of type `annotations.Annotations`, which contains both warning and info annotations.

When the result is serialized as JSON, that single field becomes two separate `warnings` and `infos` fields.

Reply:
Ah, okay, thanks!